### PR TITLE
Websocket API: Light transaction updates & `NoSuccessNotify`

### PR DESCRIPTION
**Pull request by Centril**
# Description of Changes

Does two things:

1. Adds `TransactionUpdateLight` that consists only of the `request_id` and the `DatabaseUpdate<F>`. This is emitted when the query param `&light=true` is set (unless the update is sent to the caller in which case a full update is sent).
2. Adds `flags: enum CallReducerFlags { FullUpdate, NoSuccessNotify }` to `ClientMessage`. When `NoSuccessNotify` is set, the host will not send back an empty update to the caller on success without the caller having subscribed to any affected queries. This is exposed in the SDK via `ctx.set_reducer_flags.my_reducer(CallReducerFlags::NoSuccessNotify)`. When the default `FullUpdate` is used, full updates are sent, whether the caller was subscribed to the relevant queries or not.

TODO:
- [x] Host implementation of the above.
- [x] Rust SDK implement of the above.
- [x] C# SDK implementation of the above, https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk/pull/170
- [x] TS SDK implementation of the above, https://github.com/clockworklabs/spacetimedb-typescript-sdk/pull/110

# API and ABI breaking changes

This breaks the websocket API/protocol.

### CODE REVIEW

**Comment by bfops** [comment]
@Centril I'm happy to review the CLI changes - Do you have recommendations on how to test this? Should I just regenerate bindings for an example client for each language?

File: crates/core/src/client/client_connection.rs
Comment (gefjon) [inline crates/core/src/client/client_connection.rs:270]: Does this also affect the `sender: Identity` and client address passed to the reducer in WASM? I feel like those should still be passed.

**Review by gefjon — COMMENTED** [review]
Ideally I would like to see tests added to the Rust SDK test suite which exercise this new functionality. If time is a concern, these could be added in a follow-up. I am happy to assist in writing the tests, but I don't think I have a strong enough grasp of the intended semantics to actually write them myself.

I would very much like clarification on the `ReducerContext.sender` passed to reducers whose callers requested no-notify.

I've also left many suggestions which are all: don't document the WS protocol on user-facing parts of the SDK. The WS protocol should be abstracted away to users of the SDK.

File: crates/core/src/client/client_connection.rs
Comment (gefjon) [reply crates/core/src/client/client_connection.rs:270]: Excellent, thanks!

**Review by gefjon — APPROVED** [review]

File: crates/cli/src/subcommands/subscribe.rs
Comment (bfops) [inline crates/cli/src/subcommands/subscribe.rs:208]: nit: this name doesn't seem to match the purpose of this constant

**Review by bfops — APPROVED** [review]
(codeowner review) the `crates/cli/src/subcommands/subscribe.rs` changes look benign to me :+1:

File: crates/client-api/src/routes/subscribe.rs
Comment (cloutiertyler) [inline crates/client-api/src/routes/subscribe.rs:51]: I thought the plan was to encode this as part of the protocol? Something like `v1.bin.spacetimedb.light` or am I misremembering? I could definitely be misremembering.

**Review by cloutiertyler — APPROVED** [review]
In principle, this looks good and correct to me. I just want to be cautious about merging, particularly if we have not yet tested this. Maybe we hold off on doing a basic CircleGame test tomorrow before merging.

File: crates/client-api/src/routes/subscribe.rs
Comment (cloutiertyler) [reply crates/client-api/src/routes/subscribe.rs:51]: Ack. It's definitely a bikeshed. Just wondering if you knew what we had originally said.

**Pull request merged**
